src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -5015,5 +5015,51 @@ public void M(object o)
             var compilation = CreateCompilation(source, options: TestOptions.ReleaseExe);
             base.CompileAndVerify(compilation, expectedOutput: expectedOutput);
         }
+
+        [Fact, WorkItem(19831, "https://github.com/dotnet/roslyn/issues/19831")]
+        public void CaptureAssignedInOuterFinally()
+        {
+            var source = @"
+
+    using System;
+    using System.Threading.Tasks;
+
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            Test().Wait();
+            System.Console.WriteLine(""success"");
+        }
+
+        public static async Task Test()
+        {
+            // declaring variable before try/finally and nulling it in finally cause NRE in try's body
+            var obj = new Object();
+
+            try
+            {
+                for(int i = 0; i < 3; i++)
+                {
+                    // NRE on second iteration
```

Comment:
I don't see how this is validated in the test.

Reply:
NRE is what is happening when bug reproduces. If we get expected output, there is no bug.

---
In reply to: [151242018](https://github.com/dotnet/roslyn/pull/23172#discussion_r151242018) [](ancestors = 151242018)